Tests/CommunicationLayer/CommDeviceControl/UdpMessageForwarderTest.cpp

```diff
@@ -0,0 +1,203 @@
+ */
+
+#include <gtest/gtest.h>
+#include <gmock/gmock.h>
+
+#include <QScopedPointer>
+#include <QDebug>
+#include <QByteArray>
+#include <string>
+
+#include <SimpleAmqpClient/SimpleAmqpClient.h>
+
+#include "CommunicationLayer/CommDeviceControl/UdpMessageForwarder.h"
+#include "InfrastructureLayer/Settings/MockSettings.h"
+
+using ::testing::Return;
+
+namespace
+{
+    const QString MOCK_IP = QString("localhost");
+    const quint16 MOCK_PORT = 5672;
+    const QString MOCK_QUEUE = "test-queue";
+    const QString MOCK_EXCHANGE = "amq.direct";
+    const QString MOCK_ROUTING_KEY = "routing-key";
+    const QString EXPECTED_1 = "Message Test";
+    const QString EXPECTED_2 = "Second Message Test";
+    const QString EXPECTED_3 = "Sp3c1a1** Ch4r4c74r5__ 7357";
+}
+
+class UdpMessageForwarderTest : public ::testing::Test
+{
+protected:
+
+    QScopedPointer<MockSettings> settings_;
+    QScopedPointer<UdpMessageForwarder> forwarder;
+    AmqpClient::Channel::ptr_t receiver;
+
+    // TODO test that disconnecting the sender does not destroy messages in the queue
+
+    /**
+   * @brief SetUp will set up the receiver to verify messages are being sent, as well as mocking the settings to be used by the UdpMessageForwarder
+   */
+    virtual void SetUp()
+    {
+        settings_.reset(new MockSettings());
+        receiver = Channel::Create(MOCK_IP.toStdString(), MOCK_PORT);
+        receiver->DeclareQueue(MOCK_QUEUE.toStdString(), false, true, false, false);
+
+        receiver->BindQueue(MOCK_QUEUE.toStdString(), MOCK_EXCHANGE.toStdString(), MOCK_ROUTING_KEY.toStdString());
+
+        EXPECT_CALL(*settings_, ipAddress())
+                .WillRepeatedly(Return(MOCK_IP));
+        EXPECT_CALL(*settings_, queueName())
+                .WillRepeatedly(Return(MOCK_QUEUE));
+        EXPECT_CALL(*settings_, udpPort())
+                .WillRepeatedly(Return(MOCK_PORT));
+        EXPECT_CALL(*settings_, routingKey())
+                .WillRepeatedly(Return(MOCK_ROUTING_KEY));
+    }
+
+    virtual void TearDown() {
+        receiver->PurgeQueue(MOCK_QUEUE.toStdString());
+        receiver->DeleteQueue(MOCK_QUEUE.toStdString());
+        receiver.reset();
+    }
+
+    // Test Methods
+
+    void sendMessage(const QString& message, bool resetForwarder);
+    void receiveMessage(QString& actual, bool setupConsume);
+    /**
+     * @brief setup the receiver for usage
+     */
+    void setupReceiver();
+    /**
+     * @brief reset the receiver (clear and delete queue)
+     */
+    void cleanReceiver();
+};
+
+void UdpMessageForwarderTest::sendMessage(const QString& message, bool resetForwarder) {
+    if (resetForwarder)
+        forwarder.reset(new UdpMessageForwarder(*settings_));
+    QByteArray expectedBytes = QByteArray();
+    expectedBytes.append(message);
+    try {
+        forwarder->forwardData(expectedBytes);
+    } catch (quint16 err) {
+        qDebug() << "Failed to forward data with exception: " << err;
+    }
```

Comment:
No need to have a try catch here, it's fine and less verbose to just let the forwarder throw an exception and crash the test.